### DISCUSSION TITLE
Geospatial query support for site, channel, and source

### DISCUSSION THREAD

**Discussion by pavlis** [mspass-team/mspass#126] [Design & Development]
Working on documentation I realized yet another hole in Database.  We need a method in Database to build a geospatial index for the collections listed in the title.   The geospatial syntax in mongo is a bit obscure and not something most of our user's would need to know except maybe forms for a geospatil query.   The main method we need in Database is a function to build a geospatial index for a collection.  Here is a  prototype signature:
```
def build_geospatial_index(db,collection,other_args):
```
I list *other_args* as ambiguous because how to build the index is not clear to me a this point.  It may require some regularization of names in collections.

**Comment by wangyinz** [comment]
hmmm... I totally ignored that we need to support the geospatial index when working out his version of the schema... The first issue I can think of is how we should handle the discrepancy between the data type in the database and the one in the Metadata. We will either have to make the C++ code support the GeoJSON type, or make sure any read/save of the geospatial data to the database is handled by our custom methods. It seems latter is more feasible but less elegant...

**Comment by pavlis** [comment]
Yes, that hole is why I posted this discussion.  Actually I think there is a fairly simple solution we can ultimately implement.  That is, the geospatial queries actually require  different attribute that structures lat,lon in a pair (either a list or an array - not sure which - might even be either).  I actually think the fix to pieces that handle obspy Inventory and Catalog to add the special structures needed for geospatial queries.   The only part that would take some work is a simplified query mechanism - necessary because the query structure is even weirder than regular mongodb queries.
